plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/TestMySqlTypeMapping.java

```diff
@@ -862,6 +863,110 @@ public void testTimestampFromTrino(ZoneId sessionZone)
+                .addRoundTrip("timestamp(6) WITH TIME ZONE", "TIMESTAMP '1986-01-01 00:13:07.000000 %s'".formatted(sessionZone), createTimestampWithTimeZoneType(6), "TIMESTAMP '1986-01-01 00:13:07.000000 UTC'")
+
+                // test arbitrary time for all supported precisions
+                .addRoundTrip("timestamp(0) WITH TIME ZONE", "TIMESTAMP '1970-01-01 00:00:01 %s'".formatted(sessionZone), createTimestampWithTimeZoneType(0), "TIMESTAMP '1970-01-01 00:00:01 UTC'")
+                .addRoundTrip("timestamp(1) WITH TIME ZONE", "TIMESTAMP '1970-01-01 00:00:01.1 %s'".formatted(sessionZone), createTimestampWithTimeZoneType(1), "TIMESTAMP '1970-01-01 00:00:01.1 UTC'")
+                .addRoundTrip("timestamp(1) WITH TIME ZONE", "TIMESTAMP '1970-01-01 00:00:01.9 %s'".formatted(sessionZone), createTimestampWithTimeZoneType(1), "TIMESTAMP '1970-01-01 00:00:01.9 UTC'")
+                .addRoundTrip("timestamp(2) WITH TIME ZONE", "TIMESTAMP '1970-01-01 00:00:01.12 %s'".formatted(sessionZone), createTimestampWithTimeZoneType(2), "TIMESTAMP '1970-01-01 00:00:01.12 UTC'")
+                .addRoundTrip("timestamp(3) WITH TIME ZONE", "TIMESTAMP '1970-01-01 00:00:01.123 %s'".formatted(sessionZone), createTimestampWithTimeZoneType(3), "TIMESTAMP '1970-01-01 00:00:01.123 UTC'")
+                .addRoundTrip("timestamp(3) WITH TIME ZONE", "TIMESTAMP '1970-01-01 00:00:01.999 %s'".formatted(sessionZone), createTimestampWithTimeZoneType(3), "TIMESTAMP '1970-01-01 00:00:01.999 UTC'")
+                .addRoundTrip("timestamp(4) WITH TIME ZONE", "TIMESTAMP '1970-01-01 00:00:01.1234 %s'".formatted(sessionZone), createTimestampWithTimeZoneType(4), "TIMESTAMP '1970-01-01 00:00:01.1234 UTC'")
+                .addRoundTrip("timestamp(5) WITH TIME ZONE", "TIMESTAMP '1970-01-01 00:00:01.12345 %s'".formatted(sessionZone), createTimestampWithTimeZoneType(5), "TIMESTAMP '1970-01-01 00:00:01.12345 UTC'")
+                .addRoundTrip("timestamp(1) WITH TIME ZONE", "TIMESTAMP '2020-09-27 12:34:56.1 %s'".formatted(sessionZone), createTimestampWithTimeZoneType(1), "TIMESTAMP '2020-09-27 12:34:56.1 UTC'")
+                .addRoundTrip("timestamp(1) WITH TIME ZONE", "TIMESTAMP '2020-09-27 12:34:56.9 %s'".formatted(sessionZone), createTimestampWithTimeZoneType(1), "TIMESTAMP '2020-09-27 12:34:56.9 UTC'")
+                .addRoundTrip("timestamp(3) WITH TIME ZONE", "TIMESTAMP '2020-09-27 12:34:56.123 %s'".formatted(sessionZone), createTimestampWithTimeZoneType(3), "TIMESTAMP '2020-09-27 12:34:56.123 UTC'")
+                .addRoundTrip("timestamp(3) WITH TIME ZONE", "TIMESTAMP '2020-09-27 12:34:56.999 %s'".formatted(sessionZone), createTimestampWithTimeZoneType(3), "TIMESTAMP '2020-09-27 12:34:56.999 UTC'")
+                .addRoundTrip("timestamp(6) WITH TIME ZONE", "TIMESTAMP '2020-09-27 12:34:56.123456 %s'".formatted(sessionZone), createTimestampWithTimeZoneType(6), "TIMESTAMP '2020-09-27 12:34:56.123456 UTC'")
```

Comment:
also test values where rounding needs to be performed - important to test since write path is different for CTAS and INSERT.

Reply:
Added `testTimestampWithTimeZoneCoercion`, a copy of `testTimestampCoercion` but slightly modified to to use values that are within range of MySQL TIMESTAMP